Fix #44: Always present the (at most) top 10 rows of the sheet in range selection

alaric-rd opened the pull request:
Previously, we were selecting rows based on the currently selected range, which was OK when the default initial header range was selected, but problematic when the user had already selected a header range.

Resolves #44.

rossjones commented on lib/importer/sheets.js:
```diff
@@ -32,13 +32,12 @@ exports.GetRows = (session, start=0, count=10) => {
-    // the header range to be.
-    const hRange = session.headerRange;
+    // Try to provide rows from start to start+count-1, but limit ourselves to
+    // the rows in the sheet.
```
The previous version used session.headerRange which might be the default, but also could be a selection in the middle of the sheet.  Do we handle that case another way now without using the headerRange range in the session?

alaric-rd replied:
This is used for the screen where the user selects the header rows, so it needs to display the whole sheet. When you hit it normally, the headerRange is set to a default that means you get to see everything, but the bug this tries to fix is that if you select a header range then navigate back to the selection page and reload it, it then shows less than the entire sheet because there's now a non-default headerRange set. So you can't move the header range back UP if you move it too far down by accident.

rossjones replied:
Awesome, thanks for clarifying

simonwo commented:
@alaric-rd what's happening with this PR? It seems that #44 is still an issue.

alaric-rd commented:
> @alaric-rd what's happening with this PR? It seems that #44 is still an issue.

> @alaric-rd what's happening with this PR? It seems that #44 is still an issue.

I'd forgotten about it! I've addressed @rossjones 's question and rebased it to main. Are we OK to merge it?
